system_metrics_collector/test/system_metrics_collector/test_linux_cpu_measurement.cpp

```diff
@@ -320,43 +198,98 @@ TEST_F(LinuxCpuMeasurementTestFixture, TestPublishMetricsMessage)
   //
-  test_measure_linux_cpu_->activate();
   ASSERT_TRUE(test_measure_linux_cpu_->IsStarted());
-  ASSERT_EQ(State::PRIMARY_STATE_ACTIVE, test_measure_linux_cpu_->get_current_state().id());
+  ex.spin_until_future_complete(
+    test_measure_linux_cpu_->GetFuture(), test_constants::kSpinTimeout);
 
-  ex.spin_until_future_complete(dummy_future, test_constants::kTestDuration);
-  EXPECT_EQ(6, test_receive_measurements->GetNumReceived());
-  // expectation is:
-  // 50 ms: kProcSamples[5] is collected
-  // 80 ms: statistics derived from kProcSamples[N/A-5] is published. statistics are cleared
-  // 100 ms: kProcSamples[6] is collected
-  // 150 ms: kProcSamples[7] is collected
-  // 160 ms: statistics derived from kProcSamples[5-6 & 6-7] is published. statistics are cleared
-  // 200 ms: kProcSamples[8] is collected
-  // 240 ms: statistics derived from kProcSamples[7-8] is published. statistics are cleared
-  // 250 ms: kProcSamples[9] is collected. last GetStatisticsResults() is of kProcSamples[8-9]
   data = test_measure_linux_cpu_->GetStatisticsResults();
+
   expected_cpu_active = ComputeCpuActivePercentage(
-    ProcessStatCpuLine(kProcSamples[8]),
-    ProcessStatCpuLine(kProcSamples[9]));
+    ProcessStatCpuLine(kProcSamples[2]),
+    ProcessStatCpuLine(kProcSamples[3]));
+
+  EXPECT_EQ(1, data.sample_count);
   EXPECT_DOUBLE_EQ(expected_cpu_active, data.average);
   EXPECT_DOUBLE_EQ(expected_cpu_active, data.min);
   EXPECT_DOUBLE_EQ(expected_cpu_active, data.max);
-  EXPECT_DOUBLE_EQ(0, data.standard_deviation);
-  EXPECT_EQ(1, data.sample_count);
+  EXPECT_DOUBLE_EQ(0.0, data.standard_deviation);  // only one sample so 0
+}
+
+/**
+ * Test receipt of a single published message. Expectation is 3 periodic measurements
+ * will be made and one message sent.
+ */
+TEST_F(LinuxCpuMeasurementTestFixture, TestPublishMessage)
+{
+  ASSERT_NE(test_measure_linux_cpu_, nullptr);
+  ASSERT_FALSE(test_measure_linux_cpu_->IsStarted());
+  ASSERT_EQ(
+    State::PRIMARY_STATE_UNCONFIGURED,
+    test_measure_linux_cpu_->get_current_state().id());
+
+  const auto test_receive_measurements = std::make_shared<test_functions::MetricsMessageSubscriber>(
+    "test_receive_measurements");
+
+  rclcpp::executors::SingleThreadedExecutor ex;
+  ex.add_node(test_measure_linux_cpu_->get_node_base_interface());
+  ex.add_node(test_receive_measurements);
+
+  // configure the node manually (lifecycle transition)
+  test_measure_linux_cpu_->configure();
+  ASSERT_EQ(State::PRIMARY_STATE_INACTIVE, test_measure_linux_cpu_->get_current_state().id());
+  ASSERT_FALSE(test_measure_linux_cpu_->IsStarted());
+
+  // activate the node manually (lifecycle transition): this allows collection and data publication
+  test_measure_linux_cpu_->activate();
+  ASSERT_EQ(State::PRIMARY_STATE_ACTIVE, test_measure_linux_cpu_->get_current_state().id());
+  ASSERT_TRUE(test_measure_linux_cpu_->IsStarted());
+
+  //
```

Comment:
Extra `//` for this comment throughout.

Reply:
That was from the original test, didn't change it.